src/software/power_board/chicker.cpp

```diff
@@ -0,0 +1,36 @@
+#include <Arduino.h>
+
+hw_timer_t * timer = NULL; 
+
+const int GPIO_PIN = 18;
+
+/**
+ * Creates a pulse for a given duration.
+ *
+ * @param duration the duration of the pulse in microseconds.
+ */
+void oneShotPulse(int duration) {
+  timerWrite(timer, 0);
+  timerAlarmWrite(timer, duration, false); 
+  timerAlarmEnable(timer);
+
+  digitalWrite(GPIO_PIN, HIGH);
+}
+
+/**
+ * Helper function that ends a pulse.
+ */
+void IRAM_ATTR stopPulse() {
+  digitalWrite(GPIO_PIN, LOW);
+}
+
+void setup() {
```

Comment:
I'm wondering if this should be called setupChicker so it can be called by the main setup function in the Arduino program, but I'm not familiar with the firmware architecture

Reply:
Yeah we need to cleanup all this stuff once we have all the functionality (which is why this is targetting the esp32_power_board branch)

The goal is to prototype all the peripherals and get working code

And then we can organize the code once we have all the functional peices